src/main/java/org/folio/qm/controller/ErrorHandling.java

```diff
@@ -27,11 +27,16 @@
   @ExceptionHandler(FeignException.class)
   public Error handleFeignStatusException(FeignException e, HttpServletResponse response) {
     var status = e.status();
-    var message = e.responseBody()
-      .map(byteBuffer -> new String(byteBuffer.array(), UTF_8))
-      .orElse(StringUtils.EMPTY);
-    response.setStatus(status);
-    return buildError(status, FOLIO_EXTERNAL_OR_UNDEFINED, message);
+    if (status != -1) {
+      var message = e.responseBody()
+        .map(byteBuffer -> new String(byteBuffer.array(), UTF_8))
+        .orElse(StringUtils.EMPTY);
+      response.setStatus(status);
+      return buildError(status, FOLIO_EXTERNAL_OR_UNDEFINED, message);
+    } else {
+      response.setStatus(HttpStatus.BAD_REQUEST.value());
```

Comment:
Maybe better to use REQUEST_TIMEOUT instead of BAD_REQUEST

Reply:
I assume that there could be other cases, not only request timeout.